src/main/java/de/tum/in/www1/artemis/web/rest/ModelingExerciseResource.java

```diff
@@ -171,6 +171,13 @@ public ModelingExerciseResource(ModelingExerciseRepository modelingExerciseRepos
         exerciseService.validateGeneralSettings(modelingExercise);
 
         ModelingExercise modelingExerciseBeforeUpdate = modelingExerciseRepository.findByIdElseThrow(modelingExercise.getId());
+
+        // Forbid changes of course exercise belongs to.
+        if (modelingExerciseBeforeUpdate.getCourseViaExerciseGroupOrCourseMember().getId().longValue() != modelingExercise.getCourseViaExerciseGroupOrCourseMember().getId()
+                .longValue()) {
```

Comment:
Why do we have to compare the `longValues` here instead of comparing the ids directly? (Which are `Long`s instead of `long`s I think?)

Reply:
Nvm, I see now that we can't compare objects with `==`... thanks Java.

Reply:
please use `!Object.equals(modelingExerciseBeforeUpdate.getCourseViaExerciseGroupOrCourseMember().getId(), modelingExercise.getCourseViaExerciseGroupOrCourseMember().getId())`